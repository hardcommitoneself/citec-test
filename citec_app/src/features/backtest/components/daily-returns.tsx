// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { Ts } from 'features/utils/interfaces/backtest-api-resp'
import moment from 'moment'
import ReactApexChart from 'react-apexcharts'

interface DailyReturnsProps {
   ts_backtest: Ts
}

export const DailyReturns = ({ ts_backtest }: DailyReturnsProps) => {
   const { benchmark, portfolio, date } = ts_backtest

   const portfolioConverted = portfolio.map((num) => Number((num * 100).toFixed(3)))
   const benchmarkConverted = benchmark.map((num) => Number((num * 100).toFixed(3)))
   // Y AXIS

   const series = [
      {
         name: 'Portfolio',
         data: portfolioConverted,
      },
      {
         name: 'Benchmark',
         data: benchmarkConverted,
      },
   ]

   // const type = "line" as const;

   const options: ApexCharts.ApexOptions = {
      chart: {
         type: 'line' as const,
         height: 350,
         background: '#f6f7fb',
      },
      dataLabels: {
         enabled: false,
      },
      labels: date,
      stroke: {
         curve: 'straight' as const,
         width: 1.5,
      },
      grid: {
         row: {
            colors: ['#f3f3f3', 'transparent'], // takes an array which will be repeated on columns
            opacity: 0.5,
         },
      },
      xaxis: {
         tickAmount: 11,
         categories: [],
         labels: {
            rotate: 0,
            formatter(value) {
               const formatedDate = moment(value, 'DD-MM-YYYY').format('MM-YYYY')

               return formatedDate
            },
         },
      },
      yaxis: {
         tickAmount: undefined, // apexcharts.com/docs/options/yaxis/
         floating: false,
         decimalsInFloat: 0,
      },
   }

   return (
      <ReactApexChart options={options} series={series} type="line" height={350} />
   )
}
