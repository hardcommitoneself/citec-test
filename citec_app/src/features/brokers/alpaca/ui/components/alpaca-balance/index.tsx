// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { useDispatch } from 'react-redux'
import { LinkAccount } from 'features/brokers/link-account'
import MenuItem from '@mui/material/MenuItem'
import FormControl from '@mui/material/FormControl'
import Select, { SelectChangeEvent } from '@mui/material/Select'
import { setAccountTypeSelect } from 'store/slices/brokers'
import { useAppSelector } from 'store/hooks'
import { HistoryPortfolio } from './alpaca-history-portfolio'

export const AlpacaBalance = () => {
   const visibility = process.env.REACT_APP_ALPACA_VISIBILITY

   const dispatch = useDispatch()

   const brokers = useAppSelector((state) => state.brokers)
   const typesAccount = brokers.accountType

   const handleChange = (event: SelectChangeEvent) => {
      const typeSelected = event.target.value as string
      dispatch(setAccountTypeSelect({ accountTypeSelect: typeSelected }))
   }

   return (
      <>
         {visibility == 'true' && (
            <div className="mx-10 mt-2 bg-white p-4 rounded-2xl relative balance">
               <div className="brokerTitle">
                  <div className="actionBroker">
                     <div className="selectAccount">
                        <span className="text-sm">Account</span>
                        <FormControl
                           variant="standard"
                           sx={{ m: 1, width: 200, pt: 0 }}
                           size="small"
                        >
                           <Select
                              value={brokers.accountTypeSelect}
                              label="type-acount"
                              onChange={handleChange}
                              inputProps={{ 'aria-label': 'Without label' }}
                              className="selectAccountType"
                           >
                              {typesAccount.map((typeAc) => (
                                 <MenuItem value={typeAc} key={typeAc}>
                                    {typeAc == 'PAPER'
                                       ? 'Paper Trading'
                                       : 'Live Trading'}
                                 </MenuItem>
                              ))}
                           </Select>
                        </FormControl>
                     </div>

                     <LinkAccount />
                  </div>
               </div>
               <HistoryPortfolio />
            </div>
         )}
      </>
   )
}
