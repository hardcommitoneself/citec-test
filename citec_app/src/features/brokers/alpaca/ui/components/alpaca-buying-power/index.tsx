// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { useState, useEffect } from 'react'
import { Button } from '@mui/material'
import { formatToMoney } from 'features/utils/format-to-money'
import { AlpacaApi } from 'features/brokers/alpaca/utils/hooks/api/alpaca-api'
import { useDispatch } from 'react-redux'
import { resetBrokers } from 'store/slices/brokers'
import { useAppSelector } from 'store/hooks'

export const AlpacaBuyingPower = () => {
   const dispatch = useDispatch()

   const visibility = process.env.REACT_APP_ALPACA_VISIBILITY
   const [buyingPower, setBuyingPower] = useState(0)

   const brokers = useAppSelector((state) => state.brokers)

   useEffect(() => {
      getBuyinPower(brokers.accountTypeSelect)
      const interval = setInterval(() => {
         getBuyinPower(brokers.accountTypeSelect)
      }, 5000)

      return function cleanup() {
         clearInterval(interval)
      }
   }, [brokers.accountTypeSelect])

   const getBuyinPower = (type: string) => {
      AlpacaApi({
         type: type,
         url: 'account',
         method: 'GET',
      })
         .then(function (response) {
            setBuyingPower(response.data.buying_power)
         })
         .catch(function (error) {
            if (error.response) {
               if (error.response.status == '403') {
                  console.log(error.response.status)
                  dispatch(resetBrokers())
                  localStorage.removeItem('a1c_t')
               } else {
                  console.log(error.response.data)
                  console.log(error.response.status)
                  console.log(error.response.headers)
               }
            }
         })
   }

   return (
      <>
         {visibility == 'true' && (
            <div className="mx-10 mt-8 bg-white p-4 rounded-2xl relative balance">
               <div className="flex buying-power">
                  <h3 className="font-semibold text-[26px] text-[#2d405a]">
                     Buying Power
                  </h3>
                  <h3 className="font-semibold text-[26px] text-[#2d405a] buying-power-value">
                     {formatToMoney(Math.trunc(buyingPower))}
                  </h3>
               </div>
            </div>
         )}
      </>
   )
}
