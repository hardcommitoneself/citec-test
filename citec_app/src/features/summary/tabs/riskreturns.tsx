// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { formatToMoney } from 'features/utils/format-to-money'
import { renderPopup } from 'features/popups/risk-returns-popups'
import {
   Concentration,
   RiskReturns,
   ValueAtRisk,
} from 'features/portfolio-analisys/components'

const Group = require('assets/img/group.png')

export const Risk = ({ analysis }: any) => {
   const portfolio_analysis = analysis

   return (
      <>
         <div className="RiskReturnGrid">
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Value at risk
                     <img
                        className="moreinfo"
                        src={Group}
                        alt=""
                        onClick={() => {
                           renderPopup('valueAtRisk')
                        }}
                        data-testid="info-summary-risk-var"
                     />
                  </h1>
               </div>

               <div className="">
                  <ValueAtRisk valueatrisk={portfolio_analysis?.var} />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1 data-testid="graph-risk-returns-analisys">
                     Risk vs. return
                     <img
                        className="moreinfo"
                        src={Group}
                        alt=""
                        onClick={() => {
                           renderPopup('risk')
                        }}
                        data-testid="info-summary-risk-risk"
                     />
                  </h1>
               </div>
               <div className="">
                  <RiskReturns risks={portfolio_analysis?.risk_returns} />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Concentration
                     <img
                        className="moreinfo"
                        src={Group}
                        alt=""
                        onClick={() => {
                           renderPopup('concentration')
                        }}
                        data-testid="info-summary-risk-concentration"
                     />
                  </h1>
               </div>
               <div className="">
                  <Concentration concentration={portfolio_analysis?.concentration} />
               </div>
            </div>
            <div className="box_chart_area RiskReturnTableAreaWrapper">
               <div className="top_area">
                  <h1>
                     Top 10 companies
                     <img
                        className="moreinfo"
                        src={Group}
                        alt=""
                        onClick={() => {
                           renderPopup('top10')
                        }}
                        data-testid="info-summary-risk-top10"
                     />
                  </h1>
               </div>
               <div className="chartRiskReturn" style={{ alignItems: 'flex-start' }}>
                  <table className="RiskReturnTableArea">
                     <thead>
                        <tr>
                           <th>Ticker</th>
                           <th>Name</th>
                           <th>% of total</th>
                           <th>Investment</th>
                        </tr>
                     </thead>
                     <tbody>
                        {portfolio_analysis?.top_holdings.ticker.map(
                           (ticker: string, key: number) => (
                              <tr key={key}>
                                 <td>{ticker}</td>
                                 <td></td>
                                 <td>
                                    {(
                                       portfolio_analysis.top_holdings.weight[key] *
                                       100
                                    ).toFixed(1)}
                                    %
                                 </td>
                                 <td>
                                    {formatToMoney(
                                       portfolio_analysis.top_holdings
                                          .outstanding_balance[key]
                                    )}
                                 </td>
                              </tr>
                           )
                        )}
                     </tbody>
                  </table>
               </div>
            </div>
         </div>
      </>
   )
}
