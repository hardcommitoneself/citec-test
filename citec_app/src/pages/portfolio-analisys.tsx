// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { useDispatch } from 'react-redux'
import { useEffect, useState } from 'react'
import Button from '@mui/material/Button'

import { BacktestRes } from 'features/utils/interfaces/backtest-api-resp'

import { Loading } from 'features/ui'
import { PageContainer } from 'features/ui/page-container'
import { PortfolioAnalisysResp } from 'features/utils/interfaces/portfolio-analysis-resp'
import { PositionsResp } from 'features/utils/interfaces/positions-resp'
import { setPortfolioAnalysis, setPortfolioPositions } from 'store/slices/portfolio'
import { useAppNavigate, useFetchApi } from 'features/utils/hooks'
import { useAppSelector } from 'store/hooks'
import {
   RiskreturnTab,
   Backtest,
   OverviewTab,
   Positions,
} from 'features/portfolio-analisys/tabs'
import { savePortfolio } from 'features/utils/save-portfolio'

export const PortfolioAnalysis = () => {
   const navigate = useAppNavigate()

   const { ticker, outstanding_balance, unrealized_capital_gains } = useAppSelector(
      (state) => state.portfolio
   )

   const outstanding_balance_total = outstanding_balance?.reduce(
      (accumulator, curr) => accumulator + curr
   )

   const benchmark = useAppSelector((state) => state.rebalance.benchmark)

   const { resp, isLoading, error } = useFetchApi<PortfolioAnalisysResp>({
      endpoint: '/ai/portfolio/analysis',
      method: 'POST',
      params: {
         hist_w: 5000,
      },
      payload: {
         portfolio: 'myportfolio',
         benchmark,
         ticker,
         outstanding_balance,
         unrealized_capital_gains,
      },
   })
   const { resp: backtest } = useFetchApi<BacktestRes>({
      endpoint: '/ai/portfolio/backtest',
      method: 'POST',
      params: {
         hist_w: 365,
      },
      payload: {
         portfolio: 'myportfolio',
         benchmark,
         ticker,
         outstanding_balance,
         unrealized_capital_gains,
      },
   })

   const { resp: positions } = useFetchApi<PositionsResp>({
      endpoint: '/ai/portfolio/positions',
      method: 'POST',
      payload: {
         portfolio: 'myportfolio',
         benchmark,
         ticker,
         outstanding_balance,
         unrealized_capital_gains,
      },
   })

   const handleSavePortfolio = () => {
      savePortfolio({
         navigate,
         benchmark,
         outstanding_balance,
         ticker,
         unrealized_capital_gains,
      })
   }

   const [active, setActive] = useState('Positions')
   const dispatch = useDispatch()

   const views = ['Positions', 'Overview', 'Risk', 'Backtest']

   useEffect(() => {
      if (resp) {
         dispatch(setPortfolioAnalysis({ ...resp }))
      }
      if (positions) {
         const newArr1 = positions.map((v) => ({
            ...v,
            weight: (
               (v.outstanding_balance * 100) /
               outstanding_balance_total
            ).toFixed(2),
         }))

         dispatch(setPortfolioPositions(newArr1))
      }
   }, [resp])

   const enabledViews = ['Positions', 'Risk', 'Overview', 'Backtest']

   if (isLoading) return <Loading />

   return (
      <PageContainer error={error}>
         <div className="InvestmentBody summary dashboard_body_portfolio_analysis">
            <div className="inner_dashboard_cri">
               <div className="options_summary_buttons">
                  {views.map((view) => (
                     <button
                        key={view}
                        className={active === view ? 'active-btn' : 'null'}
                        disabled={
                           //disabled is temporal, when the functionality is ready must remove this line
                           !enabledViews.includes(view)
                        }
                        onClick={() => {
                           setActive(view)
                        }}
                        data-testid={`btn-${view.toLowerCase}`}
                     >
                        {view}
                     </button>
                  ))}
               </div>
               {resp && backtest && (
                  <>
                     {active === 'Positions' && <Positions backtest={backtest} />}
                     {active === 'Backtest' && <Backtest backtest={backtest} />}
                     {active === 'Risk' && <RiskreturnTab />}
                     {active === 'Overview' && <OverviewTab backtest={backtest} />}
                  </>
               )}
               <div className="buttons_wrapper">
                  <Button
                     className="w-28 h-12"
                     color="secondary"
                     onClick={() => navigate('goBack')}
                  >
                     <span className="text-base font-normal">Back</span>
                  </Button>

                  <Button
                     className="w-28 h-12"
                     variant="save"
                     onClick={handleSavePortfolio}
                  >
                     <span>Save</span>
                  </Button>
               </div>
            </div>
         </div>
      </PageContainer>
   )
}
