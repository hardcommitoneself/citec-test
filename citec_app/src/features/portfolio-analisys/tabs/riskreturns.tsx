// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import Swal from 'sweetalert2'
import { Concentration, RiskReturns, ValueAtRisk } from '../components'
import { formatToMoney } from 'features/utils/format-to-money'
import { Icon } from 'assets/icons'
import { useAppSelector } from 'store/hooks'

export const RiskreturnTab = () => {
   const portfolio_analysis = useAppSelector((state) => state.portfolio.analysis)

   const valueAtRiskAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         <b>Value at risk</b> quantifies the portfolio’s possible financial losses 
         (in dollars or in percentage) over a specific time frame at a 
         defined confidence level..
            </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   const riskVsReturnAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         <b>Risk vs. return</b> analysis is used to determine how close your portfolio is to 
         the efficiency frontier, compared to the benchmark. The efficiency frontier refers 
         to the highest returns at a given level of risk. A portfolio is said to be efficient 
         if no other portfolio offers higher returns for a lower or equal amount of risk. 
         Where portfolios are located on the efficiency frontier depends on the investor’s 
         degree of risk tolerance.
            </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   const concentrationAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         <b>Concentration</b> Analysis helps you understand the level of exposure to the companies in which 
         you are investing more money. The graph shows the accumulated exposure to top positions, this means 
         if you select 10 stocks in the x axis the y axis shows the percentage of the money invested in 
         the top 10 positions
            </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   const top10CompaniesAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         Here are the <b>top 10 companies</b> with higher level of investment (Money invested)
         </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   return (
      <div className="summary_content_wrapper">
         <h1>Risk and Return </h1>

         <div className="RiskReturnGrid">
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Value at risk
                     <button className="moreinfo ml-2" onClick={valueAtRiskAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>

               <div className="">
                  <ValueAtRisk valueatrisk={portfolio_analysis?.var} />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Risk vs. return
                     <button className="moreinfo ml-2" onClick={riskVsReturnAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div className="">
                  <RiskReturns risks={portfolio_analysis?.risk_returns} />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Concentration
                     <button className="moreinfo ml-2" onClick={concentrationAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div>
                  <Concentration
                     concentration={portfolio_analysis?.concentration as never}
                  />
               </div>
            </div>
            <div className="box_chart_area RiskReturnTableAreaWrapper">
               <div className="top_area">
                  <h1>
                     Top 10 companies
                     <button className="moreinfo ml-2" onClick={top10CompaniesAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div className="chartRiskReturn" style={{ alignItems: 'flex-start' }}>
                  <table className="RiskReturnTableArea">
                     <thead>
                        <th>Ticker</th>
                        <th>Name</th>
                        <th>% of total</th>
                        <th>Investment</th>
                     </thead>
                     <tbody>
                        {portfolio_analysis?.top_holdings.ticker.map(
                           (ticker: string, key: number) => (
                              <tr key={key}>
                                 <td>{ticker}</td>
                                 <td></td>
                                 <td>
                                    {(
                                       portfolio_analysis.top_holdings.weight[key] *
                                       100
                                    ).toFixed(1)}
                                    %
                                 </td>
                                 <td>
                                    {formatToMoney(
                                       portfolio_analysis.top_holdings
                                          .outstanding_balance[key]
                                    )}
                                 </td>
                              </tr>
                           )
                        )}
                     </tbody>
                  </table>
               </div>
            </div>
         </div>
      </div>
   )
}
