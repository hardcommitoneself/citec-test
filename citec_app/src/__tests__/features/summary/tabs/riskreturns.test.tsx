// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import '@testing-library/jest-dom'
import { Risk } from 'features/summary/tabs/riskreturns'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { mount } from 'enzyme'
import { ValueAtRisk } from 'features/portfolio-analisys/components'

jest.mock('react-apexcharts', () => {
   return {
      __esModule: true,
      default: () => {
         return <div />
      },
   }
})

describe('Portfolio Analysis RiskreturnTab', () => {
   const initialState = {
      portfolio: {
         ticker: ['$CASH', 'AAPL', 'MSFT', 'TSLA'],
         outstanding_balance: [1000000, 50000, 250000, 10000],
         unrealized_capital_gains: [0, 11234, -65000, -36000],
         analysis: {
            fundamentals: {
               total_assets: 4,
               total_positions: 4,
               pe: 73.739873,
               pb: 28.025954,
               dividend_yield: 0.517303,
            },
            risk: {
               sharpe: -1.002506,
               volatility: 0.068686,
               beta: 0.315348,
               alpha: -0.000072,
            },
            returns: {
               returns_1M: -0.0148,
               returns_3M: -0.0103,
               returns_6M: null,
               CAGR: -0.0575,
            },
            capital_gains: {
               outstanding_balance: 1310000,
               unrealized_cap_losses: 101000,
               unrealized_cap_gains: 11234,
               realized_cap_losses: 0,
               realized_cap_gains: 0,
               taxes_cost: 0,
               taxes_rate: 0.4,
            },
            sectors: {
               sector: ['Cash', 'Consumer Discretionary', 'Information Technology'],
               percentage: [0.763359, 0.007634, 0.229008],
            },
            factors: {
               factor: [
                  'style',
                  'size',
                  'yield',
                  'quality',
                  'volatility',
                  'momentum',
                  'liquidity',
               ],
               percentage: [
                  0.513052, 0.618264, 0.59449, 0.5672, 0.432629, 0.529742, 0.465484,
               ],
            },
            segment_matrix: {
               style: [
                  'Val',
                  'Val',
                  'Val',
                  'Bld',
                  'Bld',
                  'Bld',
                  'Gwth',
                  'Gwth',
                  'Gwth',
               ],
               size: [
                  'Large',
                  'Mid',
                  'Small',
                  'Large',
                  'Mid',
                  'Small',
                  'Large',
                  'Mid',
                  'Small',
               ],
               weight: [0, 0, 0, 19.083969, 0, 0, 4.580153, 0, 0],
            },
            top_holdings: {
               ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
               weight: [0.763359, 0.19084, 0.038168, 0.007634],
               outstanding_balance: [1000000, 250000, 50000, 10000],
               unrealized_capital_gains: [0, -65000, 11234, -36000],
            },
            concentration: {
               ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
               concentration: [76.335878, 95.419847, 99.236641, 100],
            },
            var: {
               quantile: [
                  5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85,
                  90, 95,
               ],
               value: [
                  0, 0, 0, 0, 0.010967, 0.021321, 0.030917, 0.040021, 0.04883,
                  0.0575, 0.066169, 0.074978, 0.084083, 0.093678, 0.104032, 0.115563,
                  0.129003, 0.145913, 0.170978,
               ],
            },
            risk_returns: {
               ticker: ['portfolio', 'benchmark'],
               returns: [-0.0575, -0.122369],
               volatility: [0.068686, 0.190293],
               sharpe: [0.837133, 0.643053],
            },
         },
      },
   }

   const valueatrisk = {
      quantile: [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
      value: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 3.0, 10.0],
   }

   const analysis = {
      fundamentals: {
         total_assets: 4,
         total_positions: 4,
         pe: 73.739873,
         pb: 28.025954,
         dividend_yield: 0.517303,
      },
      risk: {
         sharpe: -1.002506,
         volatility: 0.068686,
         beta: 0.315348,
         alpha: -0.000072,
      },
      returns: {
         returns_1M: -0.0148,
         returns_3M: -0.0103,
         returns_6M: null,
         CAGR: -0.0575,
      },
      capital_gains: {
         outstanding_balance: 1310000,
         unrealized_cap_losses: 101000,
         unrealized_cap_gains: 11234,
         realized_cap_losses: 0,
         realized_cap_gains: 0,
         taxes_cost: 0,
         taxes_rate: 0.4,
      },
      sectors: {
         sector: ['Cash', 'Consumer Discretionary', 'Information Technology'],
         percentage: [0.763359, 0.007634, 0.229008],
      },
      factors: {
         factor: [
            'style',
            'size',
            'yield',
            'quality',
            'volatility',
            'momentum',
            'liquidity',
         ],
         percentage: [
            0.513052, 0.618264, 0.59449, 0.5672, 0.432629, 0.529742, 0.465484,
         ],
      },
      segment_matrix: {
         style: ['Val', 'Val', 'Val', 'Bld', 'Bld', 'Bld', 'Gwth', 'Gwth', 'Gwth'],
         size: [
            'Large',
            'Mid',
            'Small',
            'Large',
            'Mid',
            'Small',
            'Large',
            'Mid',
            'Small',
         ],
         weight: [0, 0, 0, 19.083969, 0, 0, 4.580153, 0, 0],
      },
      top_holdings: {
         ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
         weight: [0.763359, 0.19084, 0.038168, 0.007634],
         outstanding_balance: [1000000, 250000, 50000, 10000],
         unrealized_capital_gains: [0, -65000, 11234, -36000],
      },
      concentration: {
         ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
         concentration: [76.335878, 95.419847, 99.236641, 100],
      },
      var: {
         quantile: [
            5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90,
            95,
         ],
         value: [
            0, 0, 0, 0, 0.010967, 0.021321, 0.030917, 0.040021, 0.04883, 0.0575,
            0.066169, 0.074978, 0.084083, 0.093678, 0.104032, 0.115563, 0.129003,
            0.145913, 0.170978,
         ],
      },
      risk_returns: {
         ticker: ['portfolio', 'benchmark'],
         returns: [-0.0575, -0.122369],
         volatility: [0.068686, 0.190293],
         sharpe: [0.837133, 0.643053],
      },
   }

   const mockStore = configureStore()
   const store = mockStore(initialState)

   test('renders Portfolio Analysis page', async () => {
      const { container } = render(
         <Provider store={store}>
            <Risk analysis={analysis} />
         </Provider>
      )
      expect(container).toMatchSnapshot()
   })

   test('Test Value and Risk component', async () => {
      const { container } = render(<ValueAtRisk valueatrisk={valueatrisk} />)
      expect(container).toMatchSnapshot()
   })

   test('Show box Risk and Return of Apexchart', () => {
      const { getByTestId } = render(
         <Provider store={store}>
            <Risk analysis={analysis} />
         </Provider>
      )
      const graph = getByTestId('graph-risk-returns-analisys')
      expect(graph).toBeTruthy()
   })

   it('click on information icons', () => {
      const component = mount(
         <Provider store={store}>
            <Risk analysis={analysis} />
         </Provider>
      )
      component.find({ 'data-testid': 'info-summary-risk-var' }).simulate('click')
      component.find({ 'data-testid': 'info-summary-risk-risk' }).simulate('click')
      component
         .find({ 'data-testid': 'info-summary-risk-concentration' })
         .simulate('click')
      component.find({ 'data-testid': 'info-summary-risk-top10' }).simulate('click')
   })
})
