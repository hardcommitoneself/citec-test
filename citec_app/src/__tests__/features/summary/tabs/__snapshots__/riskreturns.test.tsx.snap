// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Analysis RiskreturnTab Test Value and Risk component 1`] = `
<div>
  <div />
</div>
`;

exports[`Portfolio Analysis RiskreturnTab renders Portfolio Analysis page 1`] = `
<div>
  <div
    class="RiskReturnGrid"
  >
    <div
      class="box_chart_area"
    >
      <div
        class="top_area"
      >
        <h1>
          Value at risk
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-risk-var"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class=""
      >
        <div />
      </div>
    </div>
    <div
      class="box_chart_area"
    >
      <div
        class="top_area"
      >
        <h1
          data-testid="graph-risk-returns-analisys"
        >
          Risk vs. return
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-risk-risk"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class=""
      >
        <div />
      </div>
    </div>
    <div
      class="box_chart_area"
    >
      <div
        class="top_area"
      >
        <h1>
          Concentration
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-risk-concentration"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class=""
      >
        <div>
          <div />
        </div>
      </div>
    </div>
    <div
      class="box_chart_area RiskReturnTableAreaWrapper"
    >
      <div
        class="top_area"
      >
        <h1>
          Top 10 companies
          <img
            alt=""
            class="moreinfo"
            data-testid="info-summary-risk-top10"
            src="group.png"
          />
        </h1>
      </div>
      <div
        class="chartRiskReturn"
        style="align-items: flex-start;"
      >
        <table
          class="RiskReturnTableArea"
        >
          <thead>
            <tr>
              <th>
                Ticker
              </th>
              <th>
                Name
              </th>
              <th>
                % of total
              </th>
              <th>
                Investment
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                $CASH
              </td>
              <td />
              <td>
                76.3
                %
              </td>
              <td>
                $ 1,000,000
              </td>
            </tr>
            <tr>
              <td>
                MSFT
              </td>
              <td />
              <td>
                19.1
                %
              </td>
              <td>
                $ 250,000
              </td>
            </tr>
            <tr>
              <td>
                AAPL
              </td>
              <td />
              <td>
                3.8
                %
              </td>
              <td>
                $ 50,000
              </td>
            </tr>
            <tr>
              <td>
                TSLA
              </td>
              <td />
              <td>
                0.8
                %
              </td>
              <td>
                $ 10,000
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
